test/test_vwap.py

```diff
@@ -0,0 +1,370 @@
+                TradeFee(Decimal("0"))
+            ))
+            market.trigger_event(MarketEvent.BuyOrderCompleted, BuyOrderCompletedEvent(
+                market.current_timestamp,
+                limit_order.client_order_id,
+                base_currency,
+                quote_currency,
+                base_currency if config.buy_fees_asset is AssetType.BASE_CURRENCY else quote_currency,
+                base_currency_traded,
+                quote_currency_traded,
+                Decimal("0"),
+                OrderType.LIMIT
+            ))
+        else:
+            market.set_balance(quote_currency, market.get_balance(quote_currency) + quote_currency_traded)
+            market.set_balance(base_currency, market.get_balance(base_currency) - base_currency_traded)
+            market.trigger_event(MarketEvent.OrderFilled, OrderFilledEvent(
+                market.current_timestamp,
+                limit_order.client_order_id,
+                limit_order.symbol,
+                TradeType.SELL,
+                OrderType.LIMIT,
+                limit_order.price,
+                limit_order.quantity,
+                TradeFee(Decimal("0"))
+            ))
+            market.trigger_event(MarketEvent.SellOrderCompleted, SellOrderCompletedEvent(
+                market.current_timestamp,
+                limit_order.client_order_id,
+                base_currency,
+                quote_currency,
+                base_currency if config.sell_fees_asset is AssetType.BASE_CURRENCY else quote_currency,
+                base_currency_traded,
+                quote_currency_traded,
+                Decimal("0"),
+                OrderType.LIMIT
+            ))
+
+    def test_limit_buy_order(self):
+        self.clock.add_iterator(self.limit_buy_strategy)
+
+        # test whether number of orders is one at start
+        # check whether the order is buy
+        # check whether the price is correct
+        # check whether amount is correct
+        order_time_1 = self.start_timestamp + self.clock_tick_size
+        self.clock.backtest_til(order_time_1)
+        self.assertEqual(1, len(self.limit_buy_strategy.active_bids))
+        bid_order: LimitOrder = self.limit_buy_strategy.active_bids[0][1]
+        self.assertEqual(Decimal("99"), bid_order.price)
+        self.assertLessEqual(bid_order.quantity, 100)
+
+        # Simulate market fill for limit buy and limit sell
+        self.simulate_limit_order_fill(self.market, bid_order)
+
+        fill_events = self.maker_order_fill_logger.event_log
+        self.assertEqual(1, len(fill_events))
+        bid_fills: List[OrderFilledEvent] = [evt for evt in fill_events if evt.trade_type is TradeType.BUY]
+        self.assertEqual(1, len(bid_fills))
+
+        # test whether number of orders is two after time delay
+        # check whether the order is buy
+        # check whether the price is correct
+        # check whether amount is correct
+        if bid_order.quantity < 100:
```

Comment:
This should be an assert rather than an if statement. If the bid order size is incorrect... then if statement would just skip it and not print an error during unit testing.